tests/core/process.c

```diff
@@ -99,19 +112,109 @@ PROCESS_TESTCASE (process_modules)
 
 PROCESS_TESTCASE (process_ranges)
 {
-  TestForEachContext ctx;
+  {
+    TestForEachContext ctx;
 
-  ctx.number_of_calls = 0;
-  ctx.value_to_return = TRUE;
-  gum_process_enumerate_ranges (GUM_PAGE_RW, range_found_cb, &ctx);
-  g_assert_cmpuint (ctx.number_of_calls, >, 1);
+    ctx.number_of_calls = 0;
+    ctx.value_to_return = TRUE;
+    gum_process_enumerate_ranges (GUM_PAGE_RW, range_found_cb, &ctx);
+    g_assert_cmpuint (ctx.number_of_calls, >, 1);
 
-  ctx.number_of_calls = 0;
-  ctx.value_to_return = FALSE;
-  gum_process_enumerate_ranges (GUM_PAGE_RW, range_found_cb, &ctx);
-  g_assert_cmpuint (ctx.number_of_calls, ==, 1);
+    ctx.number_of_calls = 0;
+    ctx.value_to_return = FALSE;
+    gum_process_enumerate_ranges (GUM_PAGE_RW, range_found_cb, &ctx);
+    g_assert_cmpuint (ctx.number_of_calls, ==, 1);
+  }
+
+  {
+    TestRangeContext ctx;
+    const size_t malloc_buf_size = 100;
+    void *malloc_buf;
+    const size_t stack_buf_size = 50;
+    unsigned char stack_buf[stack_buf_size];
+
+    malloc_buf = malloc (malloc_buf_size);
+
+    ctx.range.base_address = (GumAddress) malloc_buf;
+    ctx.range.size = malloc_buf_size;
+    ctx.found = FALSE;
+    ctx.found_exact = FALSE;
+    gum_process_enumerate_ranges (GUM_PAGE_RW, range_check_cb, &ctx);
+    g_assert (ctx.found);
+
+    ctx.range.base_address = ((GumAddress) malloc_buf) + 1;
+    ctx.range.size = malloc_buf_size - 1;
+    ctx.found = FALSE;
+    ctx.found_exact = FALSE;
+    gum_process_enumerate_ranges (GUM_PAGE_RW, range_check_cb, &ctx);
+    g_assert (ctx.found);
+
+    free (malloc_buf);
+
+    ctx.range.base_address = (GumAddress) stack_buf;
+    ctx.range.size = stack_buf_size;
+    ctx.found = FALSE;
+    ctx.found_exact = FALSE;
+    gum_process_enumerate_ranges (GUM_PAGE_RW, range_check_cb, &ctx);
+    g_assert (ctx.found);
+  }
 }
 
+#ifdef HAVE_DARWIN
+PROCESS_TESTCASE (process_malloc_ranges)
+{
+  {
+    TestForEachContext ctx;
+
+    ctx.number_of_calls = 0;
+    ctx.value_to_return = TRUE;
+    gum_process_enumerate_malloc_ranges (malloc_range_found_cb, &ctx);
+    g_assert_cmpuint (ctx.number_of_calls, >, 1);
+
+    ctx.number_of_calls = 0;
+    ctx.value_to_return = FALSE;
+    gum_process_enumerate_malloc_ranges (malloc_range_found_cb, &ctx);
+    g_assert_cmpuint (ctx.number_of_calls, ==, 1);
+  }
+
+  {
+    TestRangeContext ctx;
+    const size_t malloc_buf_size = 100;
+    void *malloc_buf;
+    const size_t stack_buf_size = 50;
+    unsigned char stack_buf[stack_buf_size];
```

Comment:
Would replace `unsigned char` with `guint8` here, and also `size_t` with `gsize`. The only exception is when we're in really platform-specific code and we use the same type as the native API uses.